tools/cosmovisor/internal/watchers/watcher.go

```diff
@@ -0,0 +1,52 @@
+package watchers
+
+import (
+	"context"
+	"time"
+
+	"cosmossdk.io/log"
+)
+
+// Watcher is an interface that defines a generic watcher that emits updates of type T.
+type Watcher[T any] interface {
+	// Updated returns a channel that emits updates of type T.
+	Updated() <-chan T
+}
+
+// ErrorHandler is an interface for handling errors and warnings in watchers.
+type ErrorHandler interface {
+	// Error handles an error as an error.
+	Error(msg string, err error)
+	// Warn handles an error as a warning.
+	Warn(msg string, err error)
+}
+
+type debugLoggerErrorHandler struct {
+	logger log.Logger
+}
+
+func (h *debugLoggerErrorHandler) Error(msg string, err error) {
+	h.logger.Warn(msg, "error", err)
+}
+
+func (h *debugLoggerErrorHandler) Warn(msg string, err error) {
+	h.logger.Debug(msg, "error", err)
+}
+
+// DebugLoggerErrorHandler returns an ErrorHandler that logs errors and warnings using the provided logger,
+// but downgrades errors to warnings and warnings to debug logs.
+func DebugLoggerErrorHandler(logger log.Logger) ErrorHandler {
+	return &debugLoggerErrorHandler{logger: logger}
+}
```

Comment:
Is there another implementation of this interface somewhere? I'm struggling to understand why this is useful instead of just using a logger directly.